### ISSUE TITLE
Difficult to remove thanos-sidecar

### ISSUE THREAD

**Issue by ctrahey** [prometheus-operator/kube-prometheus#928]
**What happened?**
Attempting to run without Thanos. By default I am seeing Thanos sidecar in Prometheus pods. 
I passed an empty object as the value of thanos: key in config, thanos sidecare was included in the prometheus pod.
No matter how I configured my jsonnet, the sidecar was included.

Only  when I manually removed `thanos: {}` from the prometheus CRD after rendering did I succeed. 

**Did you expect to see some different?**
Several places in the jsonnet implied that an empty thanos object might be an effective way to remove Thanos sidecar. This is not true, even though it does render out an empty obect to the prometheus CRD.

**How to reproduce it (as minimally and precisely as possible)**:
Default setup of kube-prometheus should not (IIUC) include thanos. Overriding config to thanos: {} should (IIUC) remove it.

**Environment**

* Prometheus Operator version:

    `18630ea`

* Kubernetes version information:

    1.20

* Kubernetes cluster kind:

    Bare metal, mixed architecture (X86 and RPi)

* Manifests:

```
local kp =
  (import 'kube-prometheus/main.libsonnet') +
  (import 'kube-prometheus/platforms/kubeadm.libsonnet') +
  (import 'kube-prometheus/addons/all-namespaces.libsonnet') +
  {
    values+:: {
      common+: {
        namespace: 'monitoring',
      },
      prometheus+: {
        namespaces: [],
        thanos: {},
      },
...

Which does render prometheus CRD to include spec.thanos: {}
```

**Anything else we need to know?**:
Manually removing `thanos` key from the prometheus CRD and applying solves the issue.
IMO this can be addressed by either project: 
1. In this project `kube-prometheus`, there should be a way to clearly indicate that thanos key of prometheus CRD should be omitted (not just empty)
2. In prometheus-operator project, an empty thanos: key in CRD should (IIUC) imply that thanos-sidecar should not be deployed.

**Comment by paulfantom** [comment]
Thanks for the detailed report. Can you test if the fix from #929 solves your issue?

**Comment by ctrahey** [comment]
@paulfantom Looks like it does!

```
$ diff <upstream>/manifests/prometheus-prometheus.yaml <this-fix>/manifests/prometheus-prometheus.yaml 
15,16c15
<     - apiVersion: v2
<       name: alertmanager-main
---
>     - name: alertmanager-main
63d61
<   thanos: {}
 
```
